src/triage/experiments/multicore.py

```diff
@@ -56,10 +56,22 @@ def process_train_test_tasks(self, tasks):
             run_task_with_splatted_arguments, self.model_train_tester.process_task
         )
 
-        logging.info("Starting parallel testing with %s processes", self.n_processes)
-        parallelize(partial_test, tasks, self.n_processes)
+        serial_tasks, parallel_tasks = self.model_train_tester.split_tasks_by_parallelizability(tasks)
+        # some tasks we want to run serially because they implement parallelization themselves
+        # others we want to parallelize via multiprocessing
+        logging.info("Starting parallel train/testing with %s processes", self.n_processes)
+        parallelize(partial_test, parallel_tasks, self.n_processes)
         logging.info("Cleaned up concurrent pool")
 
+        for serial_task in serial_tasks:
+            # if n_jobs is not set we want to use the whole machine with -1
+            # however if they set n_jobs we want to respect that
+            if "n_jobs" not in serial_task["train_kwargs"]["parameters"]:
+                logging.info("n_jobs not configured for serial task, "
+                             "defaulting to -1 for entire machine")
+                serial_task["train_kwargs"]["parameters"]["n_jobs"] = -1
```

Comment:
I'm guessing you don't feel it's important to create a copy of the dictionary rather than mutate it in place? (Just want to be sure it's called out.) Certainly I don't see a problem with this as it is, just noting the concern.